### PR TITLE
Adds specs for Dir.empty?

### PR DESCRIPTION
**Pull request by cadwallion**
Completes part of #473 

Specs for Feature [#10121](https://bugs.ruby-lang.org/issues/10121).

Dir.empty? returns true when param is directory with no entries, false
when param is directory with entries or a non-directory, and raises
Errno::ENOENT when param is a non-existent directory/file.

### CODE REVIEW

**Comment by esparta** [comment]
Ohhh, looks like we were working the same spec 😎 
I don't know if we should merge, or use one or another ...

**Comment by eregon** [comment]
@cadwallion Thank for your PR! As mentioned in #482, I would like to go with this one and integrate the good ideas from @esparta in this PR :)

File: core/dir/empty_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require File.expand_path('../fixtures/common', __FILE__)
+
+describe "Dir.empty?" do
+  ruby_version_is "2.4" do
```
Comment (eregon) [inline core/dir/empty_spec.rb]: Could you swap those two lines? It makes it clear that the method does not exist before Ruby 2.4.

File: core/dir/empty_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    before :all do
+      DirSpecs.create_mock_dirs
+      @empty_dir = DirSpecs.mock_dir "empty_dir"
```
Comment (eregon) [inline core/dir/empty_spec.rb]: You could simply do
```
@empty_dir = tmp("empty_dir")
```
here, the `tmp` helper will make sure the name is unique.

File: core/dir/empty_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    it "returns false for non-empty directories" do
+      result = Dir.empty? DirSpecs.mock_dir
+      result.should be_false
+    end
+
+    it "returns false for a non-directory" do
+      first_file = DirSpecs.mock_dir_files.first
+      result = Dir.empty? "#{DirSpecs.mock_dir}/#{first_file}"
+      result.should be_false
```
Comment (eregon) [inline core/dir/empty_spec.rb]: I like the idea from @esparta to use `__FILE__` and `__dir__` here and in the `it` above, which require no DirSpecs mock dirs.
Could you switch to using those? Then the `before :all` only needs to create the `@empty_dir`.

File: core/dir/empty_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    end
+
+    it "raises ENOENT for nonexistent directories" do
+      lambda { Dir.empty?(DirSpecs.nonexistent) }.should raise_error(Errno::ENOENT)
```
Comment (eregon) [inline core/dir/empty_spec.rb]: Instead of `DirSpecs.nonexistent`, you can simply use `tmp("nonexistent")` here.

**Comment by cadwallion** [comment]
It looks like it's a good thing I took a break before starting in on Pathname.empty? :)

Everything is in per your review.

**Comment by eregon** [comment]
This looks great, thank you for your contributions @cadwallion and @esparta!

**Pull request merged**